Mathlib/CategoryTheory/SmallObject/Iteration/UniqueHom.lean

```diff
@@ -0,0 +1,239 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.SmallObject.Iteration.Basic
+
+/-!
+# Uniqueness of morphisms in the category of iterations of functors
+
+Given a functor `Φ : C ⥤ C` and a natural transformation `ε : 𝟭 C ⟶ Φ`,
+we show in this file that there exists a unique morphism
+between two arbitrary objects in the category `Functor.Iteration ε j`
+when `j : J` and `J` is a well orderered set.
+
+-/
+
+universe u
+
+namespace CategoryTheory
+
+open Category Limits
+
+variable {C : Type*} [Category C] {Φ : C ⥤ C} {ε : 𝟭 C ⟶ Φ}
+  {J : Type u} [LinearOrder J] [IsWellOrder J (· < ·)] [OrderBot J] (j : J)
+
+namespace Functor
+
+namespace Iteration
+
+namespace Hom
+
+/-- The (unique) morphism between two objects in `Iteration ε ⊥` -/
+def mkOfBot (iter₁ iter₂ : Iteration ε (⊥ : J)) : iter₁ ⟶ iter₂ where
+  natTrans :=
+    { app := fun ⟨i, hi⟩ => eqToHom (by congr; simpa using hi) ≫ iter₁.isoZero.hom ≫
+        iter₂.isoZero.inv ≫ eqToHom (by congr; symm; simpa using hi)
+      naturality := fun ⟨i, hi⟩ ⟨k, hk⟩ φ => by
+        obtain rfl : i = ⊥ := by simpa using hi
+        obtain rfl : k = ⊥ := by simpa using hk
+        obtain rfl : φ = 𝟙 _ := rfl
+        simp }
+  natTrans_app_succ i hi := by simp at hi
+
+section
+
+variable {i : J} {iter₁ iter₂ : Iteration ε (wellOrderSucc i)}
+  (hi : i < wellOrderSucc i) (φ : iter₁.trunc hi.le ⟶ iter₂.trunc hi.le)
+
+/-- Auxiliary definition for `mkOfSucc`. -/
+noncomputable def mkOfSuccNatTransApp (k : J) (hk : k ≤ wellOrderSucc i) :
+    iter₁.F.obj ⟨k, hk⟩ ⟶ iter₂.F.obj ⟨k, hk⟩ :=
+  if hk' : k = wellOrderSucc i then
+    eqToHom (by subst hk'; rfl) ≫ (iter₁.isoSucc i hi).hom ≫
+      whiskerRight (φ.natTrans.app ⟨i, by rfl⟩) _ ≫ (iter₂.isoSucc i hi).inv ≫
+      eqToHom (by subst hk'; rfl)
```

Comment:
To extend data from `Iio` to `Iic`, you may want to use `piSplitLE` that I introduced [here](https://github.com/leanprover-community/mathlib4/blob/bea07173607044e35d179c483c9e46dba61ea9b6/Mathlib/Order/DirectedInverseSystem.lean#L123-L124).